sourced/ml/transformers/repo2df.py

```diff
@@ -1,33 +1,30 @@
 from sourced.ml.transformers import Transformer
 
 
-class Repo2DocFreq(Transformer):
-    NDOCS_KEY = -1, 0
+class Documents2DocFreq(Transformer):
 
-    def __init__(self, extractors, **kwargs):
+    def __init__(self, extractors, document_column='blob_id', **kwargs):
         super().__init__(**kwargs)
         self.extractors = extractors
+        self.document_column = document_column
 
     def __call__(self, rows):
-        processed = rows.flatMap(self.process_row)
+        processed = rows.flatMap(self._process_row)
         if self.explained:
             self._log.info("toDebugString():\n%s", processed.toDebugString().decode())
-        reduced = processed.countByKey()
-        ndocs = None
+        ndocs = rows.map(lambda x: x.asDict(self.document_column)).count()
+        reduced = processed.distinct().map(lambda x: ((x[0], x[1]), 1)).countByKey()
```

Comment:
`distinct` should not be called. It is the responsibility of the caller

Reply:
I need. `distinct()` here to count everything right. 
All this class has a tight connection to work with only one uast, so I need to do some tricks to make it work for any "document". Also that the reason why I can not use this special `NDOCS_KEY` to count documents anymore.